src/wp-admin/site-editor.php

```diff
@@ -34,6 +35,43 @@
 	}
 }
 
+// Validate postId and postType.
+if ( isset( $_GET['postId'] ) && $post_type_param ) {
+	switch ( $post_type_param ) {
+		case 'page':
+			$post = get_post( (int) $_GET['postId'] );
+
+			if ( null === $post || 'page' !== get_post_type( $post ) ) {
+				wp_die( __( 'Invalid page ID.' ) );
```

Comment:
To help with the user's experience, please add the following as [suggested by ](https://github.com/WordPress/wordpress-develop/pull/7188#issuecomment-2395701839)@peterwilsoncc :

```suggestion
				wp_die(
					__( 'Invalid page ID.' ),
					'',
					array(
					   'link_url'  => admin_url( 'site-editor.php' ),
					   'link_text' => 'Return to site editor',
					)
				);
				
```

Reply:
Resolved https://github.com/WordPress/wordpress-develop/pull/7188/commits/ff15ab1ecb005dabffb9e9368ba6fc8a2854297d